### ISSUE TITLE
ImageProcessingException/ArgumentOutOfRangeException when resizing image

### ISSUE THREAD

**Issue by mabead** [SixLabors/ImageSharp#696]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When resizing the attached image, the following exception is thrown:

```
SixLabors.ImageSharp.ImageProcessingException
  HResult=0x80131500
  Message=An error occurred when processing the image using ResizeProcessor`1. See the inner exception for more detail.
  Source=SixLabors.ImageSharp
  StackTrace:
   at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.CloneAndApply(Image`1 source, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.Apply(Image`1 source, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.DefaultInternalImageProcessorContext`1.ApplyProcessor(IImageProcessor`1 processor, Rectangle rectangle)
   at SixLabors.ImageSharp.Processing.DefaultInternalImageProcessorContext`1.ApplyProcessor(IImageProcessor`1 processor)
   at SixLabors.ImageSharp.Processing.ResizeExtensions.Resize[TPixel](IImageProcessingContext`1 source, Int32 width, Int32 height)
   at LANDR.Distribution.ImageProcessor.PictureProvider.<>c__DisplayClass4_0.<Resize>b__0(IImageProcessingContext`1 ctx) in C:\git\Distribution\src\LANDR.Distribution.ImageProcessor\PictureProvider.cs:line 84

Inner Exception 1:
ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
```

Here's the image:
[Sans titre 1.zip](https://github.com/SixLabors/ImageSharp/files/2357397/Sans.titre.1.zip)

### Steps to Reproduce
```csharp
using (var imageStream = File.OpenRead(imageName))
using (var image = Image.Load(imageStream))
{
   image.Mutate(ctx => ctx.Resize(640, 640));
}
```
### System Configuration

- ImageSharp version: 1.0.0-beta005
- Other ImageSharp packages and versions:
- Environment (Operating system, version and so on): windows 10
- .NET Framework version: .NET core 2.0
- Additional information:

**Comment by JimBobSquarePants** [comment]
@mabead Have you checked the EXIF profile of this image? Could you do so with this (and others) via EXIFTool.

http://owl.phy.queensu.ca/~phil/exiftool/

**Comment by JimBobSquarePants** [comment]
Looks like it's an EXIF issue. @dlemstra I may need your assistance with this and #694 

``` c#

   at System.ReadOnlySpan`1..ctor(T[] array, Int32 start, Int32 length)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.TryReadSpan(Int32 length, ReadOnlySpan`1& span) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifReader.cs:line 402
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ReadUInt16() in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifReader.cs:line 419
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.AddValues(List`1 values, Int32 index) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifReader.cs:line 159
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.GetThumbnail(Int32 offset) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifReader.cs:line 437
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ReadValues() in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifReader.cs:line 94
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.InitializeValues() in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifProfile.cs:line 295
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.get_Values() in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifProfile.cs:line 109
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.GetValue(ExifTag tag) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\MetaData\Profiles\Exif\ExifProfile.cs:line 151
   at SixLabors.ImageSharp.Processing.Processors.Transforms.TransformHelpers.UpdateDimensionalMetData[TPixel](Image`1 image) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\Processors\Transforms\TransformHelpers.cs:line 32
   at SixLabors.ImageSharp.Processing.Processors.Transforms.TransformProcessorBase`1.AfterImageApply(Image`1 source, Image`1 destination, Rectangle sourceRectangle) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\Processors\Transforms\TransformProcessorBase.cs:line 19
   at SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeProcessor`1.AfterImageApply(Image`1 source, Image`1 destination, Rectangle sourceRectangle) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\Processors\Transforms\ResizeProcessor.cs:line 374
   at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.CloneAndApply(Image`1 source, Rectangle sourceRectangle) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\Processors\CloningImageProcessor.cs:line 50
   at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.Apply(Image`1 source, Rectangle sourceRectangle) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\Processors\CloningImageProcessor.cs:line 62
   at SixLabors.ImageSharp.Processing.DefaultInternalImageProcessorContext`1.ApplyProcessor(IImageProcessor`1 processor, Rectangle rectangle) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\DefaultInternalImageProcessorContext.cs:line 73
   at SixLabors.ImageSharp.Processing.DefaultInternalImageProcessorContext`1.ApplyProcessor(IImageProcessor`1 processor) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\DefaultInternalImageProcessorContext.cs:line 80
   at SixLabors.ImageSharp.Processing.ResizeExtensions.Resize[TPixel](IImageProcessingContext`1 source, Int32 width, Int32 height, IResampler sampler, Rectangle targetRectangle, Boolean compand) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\ResizeExtensions.cs:line 173
   at SixLabors.ImageSharp.Processing.ResizeExtensions.Resize[TPixel](IImageProcessingContext`1 source, Int32 width, Int32 height, IResampler sampler, Boolean compand) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\ResizeExtensions.cs:line 120
   at SixLabors.ImageSharp.Processing.ResizeExtensions.Resize[TPixel](IImageProcessingContext`1 source, Int32 width, Int32 height) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\ResizeExtensions.cs:line 63
   at SixLabors.ImageSharp.Tests.GeneralFormatTests.<>c.<DecodeThenEncodeImageFromStreamShouldSucceed>b__2_0(IImageProcessingContext`1 ctx) in C:\development\github\SixLabors\ImageSharp\tests\ImageSharp.Tests\Formats\GeneralFormatTests.cs:line 61
   at SixLabors.ImageSharp.Processing.ProcessingExtensions.Mutate[TPixel](Image`1 source, Action`1 operation) in C:\development\github\SixLabors\ImageSharp\src\ImageSharp\Processing\ProcessingExtensions.cs:line 41
   at SixLabors.ImageSharp.Tests.GeneralFormatTests.DecodeThenEncodeImageFromStreamShouldSucceed() in C:\development\github\SixLabors\ImageSharp\tests\ImageSharp.Tests\Formats\GeneralFormatTests.cs:line 61
``

**Comment by mabead** [comment]
I haven't checked the exif profile. In fact, I have no clue what exif is so I wouldn't know what to check.

**Comment by JimBobSquarePants** [comment]
No worries, EXIF is a metadata standard used for storing supplementary data in many image formats. Our current issues are caused by a bug in the EXIFReader that I think I've just fixed.

**Comment by JimBobSquarePants** [comment]
Fixed with #699
